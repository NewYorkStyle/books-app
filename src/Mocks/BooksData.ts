import {IBook, IBooksFilter} from '../Modules/Main/Models';

export const mockBookData = [
    {
        id: '1',
        name: '20000 лье под водой',
        author: 'Жюль Верн',
        genre: 'Фантастика',
    },
    {
        id: '2',
        name: 'Убийство в "Восточном экспрессе"',
        author: 'Агата Кристи',
        genre: 'Детектив',
    },
    {
        id: '3',
        name: 'Война и мир',
        author: 'Лев Толстой',
        genre: 'Роман',
    },
    {
        id: '4',
        name: 'Пикник на обочене',
        author: 'Братья Стругацкие',
        genre: 'Фантастика',
    },
    {
        id: '5',
        name: 'Муму',
        author: 'Иван Тургенев',
        genre: 'Рассказ',
    },
    {
        id: '6',
        name: 'Повесть временных лет',
        author: 'Нестор Летописец',
        genre: 'Повесть',
    },
    {
        id: '7',
        name: 'Сказка о царе Салтане',
        author: 'Александр Пушкин',
        genre: 'Сказка',
    },
];

export const getBookData = (filter?: IBooksFilter): IBook[] => {
    let data = mockBookData;

    if (!!filter) {
        let key: keyof IBooksFilter;
        data = mockBookData.filter((item: IBook) => {
            for (key in filter) {
                if (key === 'name') {
                    return item[key].toLowerCase().indexOf(filter[key].toLowerCase()) !== -1;
                }
                if (!!filter[key] && item[key] !== filter[key]) return false;
            }
            return true;
        });
    }

    return data;
};
