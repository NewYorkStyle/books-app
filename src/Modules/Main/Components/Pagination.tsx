import {Button} from '@material-ui/core';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
import * as React from 'react';

/**
 * Компонент для отрисовки пагинации.
 *
 * @prop {number} currentlyPage Текущая страница.
 * @prop {boolean} hasNextPage Наличие следующей страницы.
 * @prop {function} onNextPageClick Обработчик нажания на следующую страницу.
 * @prop {function} onPrevPageClick Обработчик нажания на предыдущую страницу.
 */
export const Pagination: React.SFC<{
    currentlyPage: number;
    hasNextPage: boolean;
    onNextPageClick: () => void;
    onPrevPageClick: () => void;
}> = (props) => {
    const {currentlyPage, hasNextPage, onNextPageClick, onPrevPageClick} = props;
    return (
        <div className="pagination">
            <Button onClick={onPrevPageClick} disabled={currentlyPage === 1}>
                <ArrowBackIcon />
            </Button>
            <span>{currentlyPage}</span>
            <Button onClick={onNextPageClick} disabled={!hasNextPage}>
                <ArrowForwardIcon />
            </Button>
        </div>
    );
};
