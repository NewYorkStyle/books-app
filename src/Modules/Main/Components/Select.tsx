import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import MaterialSelect from '@material-ui/core/Select';
import * as React from 'react';

/**
 * Компонент для отрисовки выпадающего списка.
 *
 * @prop {string} label Лейбл.
 * @prop {function} onChange Обработчик события изменения.
 * @prop {string[]} options Данные фильтра.
 * @prop {string} value Выбраное значение фильтра.
 */
export const Select: React.SFC<{label: string; onChange: (e: any) => void; options: string[]; value: string}> = (props) => {
    const {label, onChange, options, value} = props;
    return (
        <FormControl className="filter">
            <InputLabel>{label}</InputLabel>
            <MaterialSelect value={value} onChange={onChange}>
                <MenuItem value="">
                    <em>None</em>
                </MenuItem>
                {options.map((option) => (
                    <MenuItem key={option} value={option}>
                        {option}
                    </MenuItem>
                ))}
            </MaterialSelect>
        </FormControl>
    );
};
