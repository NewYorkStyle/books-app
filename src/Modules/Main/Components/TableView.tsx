import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import * as React from 'react';

/**
 * Компонент для отрисовки таблицы.
 *
 * @prop {Array<any>} items Данные для отрисовки.
 */
export const TableView: React.SFC<{items: Array<any>}> = (props) => (
    <Table className="table" aria-label="simple table">
        <TableHead>
            <TableRow>
                <TableCell>Название книги</TableCell>
                <TableCell>Автор</TableCell>
                <TableCell>Жанр</TableCell>
            </TableRow>
        </TableHead>
        <TableBody>
            {props.items.map((book) => (
                <TableRow key={book.id}>
                    <TableCell>{book.name}</TableCell>
                    <TableCell>{book.author}</TableCell>
                    <TableCell>{book.genre}</TableCell>
                </TableRow>
            ))}
        </TableBody>
    </Table>
);
