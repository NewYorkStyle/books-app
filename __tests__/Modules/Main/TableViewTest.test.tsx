import * as React from 'react';
import {mockBookData} from '../../../src/Mocks/BooksData';
import {TableView} from '../../../src/Modules/Main/Components/TableView';

const TestRenderer = require('react-test-renderer');

it('renders correctly', () => {
    const tree = TestRenderer.create(<TableView items={mockBookData} />).toJSON();
    expect(tree).toMatchSnapshot();
});
