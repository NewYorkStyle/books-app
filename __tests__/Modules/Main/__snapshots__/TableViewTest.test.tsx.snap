// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<table
  aria-label="simple table"
  className="MuiTable-root table"
>
  <thead
    className="MuiTableHead-root"
  >
    <tr
      className="MuiTableRow-root MuiTableRow-head"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head"
        scope="col"
      >
        Название книги
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head"
        scope="col"
      >
        Автор
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head"
        scope="col"
      >
        Жанр
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root"
  >
    <tr
      className="MuiTableRow-root"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        20000 лье под водой
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Жюль Верн
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Фантастика
      </td>
    </tr>
    <tr
      className="MuiTableRow-root"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Убийство в "Восточном экспрессе"
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Агата Кристи
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Детектив
      </td>
    </tr>
    <tr
      className="MuiTableRow-root"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Война и мир
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Лев Толстой
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Роман
      </td>
    </tr>
    <tr
      className="MuiTableRow-root"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Пикник на обочене
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Братья Стругацкие
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Фантастика
      </td>
    </tr>
    <tr
      className="MuiTableRow-root"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Муму
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Иван Тургенев
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Рассказ
      </td>
    </tr>
    <tr
      className="MuiTableRow-root"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Повесть временных лет
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Нестор Летописец
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Повесть
      </td>
    </tr>
    <tr
      className="MuiTableRow-root"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Сказка о царе Салтане
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Александр Пушкин
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Сказка
      </td>
    </tr>
  </tbody>
</table>
`;
